kodi_addon_checker/check_dependencies.py

```diff
@@ -29,9 +32,15 @@ def check_addon_dependencies(report: Report, repo_addons: dict, parsed_xml, bran
 
     for required_addon, required_version in deps.items():
         if required_addon not in repo_addons:
-            if required_addon not in ignore:
+            if required_addon not in ignore and required_addon in VERSION_ATTRB:
```

Comment:
`branch_name` might also be invalid. And this is not really a nice fix, as this would just swallow the error and nobody will ever know what it happend. :/

Reply:
also this is the wrong way around I think. Best would be to extract this part of the logic into it's own method and write unit tests for this.

Reply:
How can `branch_name` might be invalid ?
I don't understand how this will swallow "error" when we won't have any error ? Can you please explain this with an example ?

> also this is the wrong way around I think. Best would be to extract this part of the logic into it's own method and write unit tests for this.

Meaning make a whole new function for this check ?

Reply:
1. It can't. We should be fine, unless we change https://github.com/xbmc/addon-check/blob/98b2954dddba6c7972e2a890c9e6f11b62235cc2/kodi_addon_checker/__main__.py#L56 and maybe add the next version. But forget about `VERSION_ATTRB` in the update process. Then we will be referencing something that doesn't exist and it will just crash, which is probably fine, but reporting errors is nicer most of the times.

2. Well if `required_addon in VERSION_ATTRB` is false we end up in the else branch. Which will just do it's thing and never mention that the required_addon was misconfigured by us somehow.

3. For this subpart, it's just a recommendation, as I've seen that you got it the wrong way around and writing tests helps figuring stuff like that out. But I would also understand if you think it's to difficult.

Branch name is just a string variable

Reply:
For 1 should report an error("There is some misconfiguration with your keys/values") in our format or should I log it ?

Reply:
Log it please, it's an error on our (addon-check) site if it happens.